diff --git a/lib/request.js b/lib/request.js
index ebf686ca5620847ff4de2ee22a5a80fb4c67ea60..c8bca375de0c5cb289acff3a99f6303d3261402c 100644
--- a/lib/request.js
+++ b/lib/request.js
@@ -220,10 +220,11 @@ Object.defineProperties(Request.prototype, {
       if (ac) return ac.signal
       ac = new AbortController()
       this[kRequestSignal] = ac
+      const raw = this.raw
       const onAbort = () => {
-        if (!ac.signal.aborted) ac.abort()
+        if (raw.aborted && !ac.signal.aborted) ac.abort()
       }
-      this.raw.on('close', onAbort)
+      raw.on('close', onAbort)
       this[kOnAbort] = onAbort
       return ac.signal
     }
diff --git a/lib/route.js b/lib/route.js
index 1cb07fa95f4508960a4fe8ed0c4cd4a337508c68..a5ad76da600a29640402a59568eb965b0e5dc7d7 100644
--- a/lib/route.js
+++ b/lib/route.js
@@ -539,8 +539,10 @@ function buildRouting (options) {
       const onAbort = () => {
-        if (!ac.signal.aborted) {
-          ac.abort()
+        if (req.aborted) {
+          if (!ac.signal.aborted) {
+            ac.abort()
+          }
+          clearTimeout(request[kTimeoutTimer])
         }
-        clearTimeout(request[kTimeoutTimer])
       }
       req.on('close', onAbort)
       request[kOnAbort] = onAbort
