diff --git a/fs-ext.cc b/fs-ext.cc
--- a/fs-ext.cc
+++ b/fs-ext.cc
@@ -95 +95,2 @@
-static void EIO_After(uv_work_t *req) {
+static void EIO_After(uv_work_t *req, int status) {
+  (void)status;
@@ -357 +358 @@
-    uv_queue_work(uv_default_loop(), req, EIO_Flock, (uv_after_work_cb)EIO_After);
+    uv_queue_work(uv_default_loop(), req, EIO_Flock, EIO_After);
@@ -427 +428 @@
-  uv_queue_work(uv_default_loop(), req, EIO_Seek, (uv_after_work_cb)EIO_After);
+  uv_queue_work(uv_default_loop(), req, EIO_Seek, EIO_After);
@@ -464 +465 @@
-  uv_queue_work(uv_default_loop(), req, EIO_Fcntl, (uv_after_work_cb)EIO_After);
+  uv_queue_work(uv_default_loop(), req, EIO_Fcntl, EIO_After);
@@ -514 +515 @@
-  uv_queue_work(uv_default_loop(), req, EIO_StatVFS,(uv_after_work_cb)EIO_After);
+  uv_queue_work(uv_default_loop(), req, EIO_StatVFS, EIO_After);
@@ -619,3 +620 @@
-  NODE_MODULE_INIT() {
-    init(exports);
-  }
+  NAN_MODULE_WORKER_ENABLED(fs_ext, init)
